### DISCUSSION TITLE
How to unmount the poopup onsubmit/onclose?

### DISCUSSION THREAD

**Discussion by vuki656** [MunifTanjim/nui.nvim#25] [Q&A] [answered]
Hey, 

I noticed that after I call `menu:mount()`, and then submit it, and the call `menu:mount()` again, it's going to open it, but it's not going to be auto-focused. I'm guessing this has something to do with it not being unmounted after `on_submit`/`on_cancel` call. 

Should the popup be unmounted after submit/cancel calls?

What to do when the popup needs to be opened/closed multiple times in the same buffer? 

Appreciate any help. 

This is the code I'm using 

```lua
local Menu = require("nui.menu")

local menu = Menu({
    relative = "cursor",
    border = {
        style = "rounded",
        highlight = "Normal",
        text = {
            top = "Delete react package",
            top_align = "left",
        },
    },
    position = {
        row = 0,
        col = 0,
    },
    size = {
        width = 50,
        height = 2,
    },
    highlight = "Normal:Normal",
    focusable = true,
}, {
    lines = {
        Menu.item("Confirm"),
        Menu.item("Cancel"),
    },
    keymap = {
        focus_next = { "j", "<Down>", "<Tab>" },
        focus_prev = { "k", "<Up>", "<S-Tab>" },
        close = { "<Esc>", "<C-c>" },
        submit = { "<CR>", "<Space>" },
    },
    on_close = function()
        print("CLOSED")
    end,
    on_submit = function(item)
        print("SUBMITTED", vim.inspect(item))
    end,
})

M.delete = function()
    menu:mount()
end
```

**Comment by MunifTanjim** [comment, marked as answer]
Hello @vuki656 , thanks for trying it out.

> I noticed that after I call menu:mount(), and then submit it, and the call menu:mount() again, it's going to open it, but it's not going to be auto-focused.

That is definitely a bug in the underlying `Popup` component. Can you confirm if https://github.com/MunifTanjim/nui.nvim/pull/27 fixes this issue for you?

**Reply by vuki656** [reply]
That seems to fix it. Thanks.
